definitions/npm/shelljs_v0.7.x/flow_>=v0.28.x/shelljs_v0.7.x.js

```diff
@@ -0,0 +1,381 @@
+  & ((pattern: $npm$shelljs$Pattern, subst: string, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdSet =
+  & ((exitOnError: '-e' | '+e', rest: void) => void)
+  & ((verbose: '-v' | '+v', rest: void) => void)
+  & ((disableGlobbing: '-f' | '+f', rest: void) => void)
+  ;
+
+declare type $npm$shelljs$OptSort = $npm$shelljs$ICmdOptions<'-n' | '-r'>;
+declare type $npm$shelljs$CmdSort =
+  & ((options: $npm$shelljs$OptSort, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  & ((glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdTail =
+  & ((num: number, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  & ((glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdTempdir =
+  & ((rest: void) => string)
+  ;
+
+declare type $npm$shelljs$OptTest = '-b' | '-c' | '-d' | '-e' | '-f' | '-L' | '-p' | '-S';
+declare type $npm$shelljs$CmdTest =
+  & ((mode: $npm$shelljs$OptTest, path: string, rest: void) => boolean)
+  ;
+
+declare type $npm$shelljs$OptTouch = {
+  [key: '-a' | '-c' | '-m' ]: true;
+  '-d'?: string;
+  '-r'?: string;
+};
+declare type $npm$shelljs$CmdTouch =
+  & ((options: $npm$shelljs$OptTouch, glob: string, ...rest: string[]) => $npm$shelljs$String)
+  & ((glob: string, ...rest: string[]) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$CmdWhich =
+  & ((cmd: string, rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeCat =
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeExec =
+  & ((cmd: string, options: $npm$shelljs$OptExec & { async: true }, callback: $npm$shelljs$FunExec, rest: void) => $npm$shelljs$Async)
+  & ((cmd: string, options: $npm$shelljs$OptExec & { async: true }, rest: void) => $npm$shelljs$Async)
+  & ((cmd: string, options: $npm$shelljs$OptExecSync, rest: void) => $npm$shelljs$String)
+  & ((cmd: string, rest: void) => $npm$shelljs$String)
+  & ((cmd: string, callback: $npm$shelljs$FunExec, rest: void) => $npm$shelljs$Async)
+  ;
+
+declare type $npm$shelljs$PipeGrep =
+  & ((options: $npm$shelljs$OptGrep, pattern: $npm$shelljs$Pattern, rest: void) => $npm$shelljs$String)
+  & ((pattern: $npm$shelljs$Pattern, rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeHead =
+  & ((num: number, rest: void) => $npm$shelljs$String)
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeSed =
+  & ((pattern: $npm$shelljs$Pattern, subst: string, rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeSort =
+  & ((options: $npm$shelljs$OptSort, rest: void) => $npm$shelljs$String)
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare type $npm$shelljs$PipeTail =
+  & ((num: number, rest: void) => $npm$shelljs$String)
+  & ((rest: void) => $npm$shelljs$String)
+  ;
+
+declare module 'shelljs' {
```

Comment:
Since it looks like this module only defined named exports, I would recommend 2 tweaks here:

1. You can switch all the `declare`s inside your `declare module` body here to be `declare export` 
2. You can then move all of the global `$npm$shelljs$*` stuff inside the `declare module` body and, because you're not using `declare export`, they will be hidden from the exports of other modules.

There was a practice of using namespaced globals in other libdefs before because there wasn't a good option of hiding `declare`s inside a `declare module` before -- but that was fixed for ES module `declare module`s in v0.28!